test-lit/lit_port.py

```diff
@@ -0,0 +1,256 @@
+    if test in TS.error_tests:
+        port_error_test(filename, new_path)
+        if test not in FAILS:
+            test_single_port(new_path, filename)
+    else:
+        print(
+            "{} could not be ported as it is not supported".format(test),
+            file=sys.stderr)
+        FAILS.append(test)
+
+
+def port_error_test(filename, savepath):
+    test = filename.name
+    options = ["-fdebug-resolve-names", "-fparse-only"]
+    lines = []
+    try:
+        with filename.open() as read_file:
+            lines = read_file.readlines()
+            for line in lines:
+                if "OPTIONS:" in line:
+                    opts = shlex.split(line)
+                    for opt in opts[1:]:
+                        options.append(opt)
+            prefixes = get_prefixes(lines)
+            if prefixes:
+                if "ERROR" in prefixes:
+                    run_line = ERROR_TEMPLATE
+                else:
+                    run_line = PREFIX_TEMPLATE
+                run_line = run_line.format(
+                    " ".join(options), ",".join(prefixes))
+            else:
+                run_line = DEFAULT_TEMPLATE.format(" ".join(options))
+
+            index = -1
+            # Add run line to untouched test
+            for line in lines:
+                # Find the first line without comment character
+                if line != "\n" and not line.startswith("!"):
+                    index = lines.index(line)
+                    break
+            lines.insert(index, "\n\n")
+            lines.insert(
+                index, run_line)
+            try:
+                with savepath.open('w') as write_file:
+                    write_file.writelines(lines)
+                print("{} completed".format(test))
+            except IOError as e:
+                print(
+                    "Could not write to {} because {}".format(
+                        savepath, e.strerror), file=sys.stderr)
+                FAILS.append(test)
+    except IOError as e:
+        print(
+            "Could not open {} because {}".format(
+                filename, e.strerror), file=sys.stderr)
+        FAILS.append(test)
+
+
+def get_prefixes(lines):
+    prefixes = set()
+    non_prefixes = ["RUN:", "OPTIONS:", "TODO:", "FIXME:", "CHECK-NOT:"]
+    for line in lines:
+        if line.strip().startswith("!") and ":" in line:
+            if "#" in line or "TODO" in line:
+                continue
+            try:
+                line = shlex.split(line)
+            except ValueError:
+                line = line.split()
+            prefix = ""
+            for token in line:
+                if token.startswith("!"):
+                    prefix = token[1:]
+                    break
+            if prefix and prefix.endswith(":") and prefix not in non_prefixes:
+                prefixes.add(prefix[:-1])
+    return prefixes
+
+
+def run(command):
+    command = "PATH=~/f18/build/install/bin:$PATH;{}".format(command)
```

Comment:
There seem to be assumptions about where the compiler and associated binaries will be. If this script is designed to be run by others then it might make sense to define this is a global variable somewhere or passed as an argument.